kolibri/core/assets/src/exams/utils.js

```diff
@@ -206,54 +220,31 @@ export function annotateQuestionSourcesWithCounter(questionSources) {
 
 // idk the best place to place this function
 export function getExamReport(examId, tryIndex = 0, questionNumber = 0, interactionIndex = 0) {
-  return new Promise((resolve, reject) => {
-    const examPromise = ExamResource.fetchModel({ id: examId });
-
-    examPromise.then(
-      exam => {
-        const questionSources = exam.question_sources;
-
-        let contentPromise;
-
-        if (questionSources.length) {
-          contentPromise = ContentNodeResource.fetchCollection({
-            getParams: {
-              ids: uniq(questionSources.map(item => item.exercise_id)),
-              no_available_filtering: true,
-            },
-          });
-        } else {
-          contentPromise = Promise.resolve([]);
-        }
-
-        contentPromise.then(
-          contentNodes => {
-            const questions = convertExamQuestionSources(exam, { contentNodes });
-
-            // When all the Exercises are not available on the server
-            if (questions.length === 0) {
-              return resolve({ exam });
-            }
-
-            const exercise = contentNodes.find(
-              node => node.id === questions[questionNumber].exercise_id
-            );
-
-            const payload = {
-              exerciseContentNodes: [...contentNodes],
-              exam,
-              questions,
-              tryIndex: Number(tryIndex),
-              questionNumber: Number(questionNumber),
-              exercise,
-              interactionIndex: Number(interactionIndex),
-            };
-            resolve(payload);
-          },
-          error => reject(error)
-        );
-      },
-      error => reject(error)
-    );
+  return ExamResource.fetchModel({ id: examId }).then(examData => {
+    return fetchExamWithContent(examData).then(({ exam, exercises }) => {
+      // When all the Exercises are not available on the server
+      if (exam.question_count === 0) {
+        return exam;
+      }
+
+      // TODO: Reports will eventually want to have the proper section-specific data to render
```

Comment:
We're not, as in within this PR, or not within this release? Won't that just give us the same extremely long question issue in reports that we have been trying to avoid in the taking of quizzes?